Bulldozer.CSV/Maps/Attendance.cs

```diff
@@ -30,207 +33,328 @@ namespace Bulldozer.CSV
             if ( archivedSchedule == null )
             {
-                archivedSchedule = AddNamedSchedule( lookupContext, archivedScheduleName, null, null, null,
+                archivedSchedule = AddNamedSchedule( rockContext, archivedScheduleName, null, null, null,
                     ImportDateTime, archivedScheduleName.RemoveSpecialCharacters(), true, ImportPersonAliasId );
             }
+            var archivedScheduleId = archivedSchedule.Id;
 
-            var completed = 0;
-            var importedCount = 0;
-            var alreadyImportedCount = attendanceService.Queryable().AsNoTracking().Count( a => a.ForeignKey != null );
-            ReportProgress( 0, string.Format( "Starting attendance import ({0:N0} already exist).", alreadyImportedCount ) );
+            var attendanceCount = attendanceData.Count;
+            HashSet<string> attendanceIds = new HashSet<string>();
 
-            string[] row;
-            // Uses a look-ahead enumerator: this call will move to the next record immediately
-            while ( ( row = csvData.Database.FirstOrDefault() ) != null )
+            attendanceData = attendanceData.Where( a => !string.IsNullOrWhiteSpace( a.GroupId ) ).ToList();
+            if ( attendanceData.Count < attendanceCount )
             {
-                var rowAttendanceKey = row[AttendanceId];
-                var rowGroupKey = row[AttendanceGroupId];
-                var rowPersonKey = row[AttendancePersonId];
-                var rowDate = row[AttendanceDate];
-                var rowCreatedDate = row[AttendanceCreatedDate];
-                var rowAttended = row[AttendanceAttended];
-                var rowLocationKey = row[AttendanceLocationId];
-                var rowAttendanceId = rowAttendanceKey.AsType<int?>();
-
-                //
-                // Find this person in the database.
-                //
-                var personKeys = GetPersonKeys( rowPersonKey );
-                if ( personKeys == null || personKeys.PersonId == 0 )
-                {
-                    ReportProgress( 0, string.Format( "Person key {0} not found", rowPersonKey ) );
-                }
+                ReportProgress( 0, string.Format( "{0:N0} attendance records have no GroupId provided and will be skipped.", attendanceCount - attendanceData.Count ) );
+                attendanceCount = attendanceData.Count;
+            }
+            attendanceData = attendanceData.Where( a => !string.IsNullOrWhiteSpace( a.PersonId ) ).ToList();
+            if ( attendanceData.Count < attendanceCount )
+            {
+                ReportProgress( 0, string.Format( "{0:N0} attendance records have no PersonId provided and will be skipped.", attendanceCount - attendanceData.Count ) );
+                attendanceCount = attendanceData.Count;
+            }
+            attendanceData = attendanceData.Where( a => a.StartDateTime.HasValue ).ToList();
+            if ( attendanceData.Count < attendanceCount )
```

Comment:
This includeds missing GroupId, PersonId, and StartDateTime. Need to make more of a distinction